### PR TITLE
Creating openshift kubetest2 installer

### PR DESCRIPTION
**Pull request by chrislovecnm**
## What this is
This new binary uses the kubetest2 interface to create
a new OpenShift cluster.

This is an out-of-tree deployer that follows https://github.com/kubernetes-sigs/kubetest2 interface.
This deployer

1. Creates an Openshift cluster on GCP
2. Deletes an Openshift cluster on GCP

## Why we need it

We are currently using kubetest2 to run integrations tests on GKE and kind, and this is another deployer that enables us to run tests on OpenShift.  We need to run e2e tests against OpenShift to certify a release.

https://github.com/cockroachdb/cockroach-operator/pull/422/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R136 is an example of how to run this binary

### TODO

- [x] Add logging
- [x] Add capability to not create a SA and reuse a token
- [x] README file with examples
- [x] Add some exception handling
- [x] Download `openshift-installer` using bazel
- [x] File issues to move this to the new GitHub Repo
- [x] Open issues to enable nightly builds
- [x] Open issue to remove the shell scripts that are basically this program
- [x] Open an issue to fix e2e tests on OpenShift
- [x] Open an issue to fix e2e test logging, add logging to get pod logs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -128,6 +128,22 @@ test/e2e/gke:
...
+
+.PHONY: test/e2e/openshift
+test/e2e/openshift:
+	bazel build //hack/bin/... //e2e/kubetest2-openshift/...
+	PATH=${PATH}:bazel-bin/hack/bin:bazel-bin/e2e/kubetest2-openshift/kubetest2-openshift_/ \
+	     bazel-bin/hack/bin/kubetest2 openshift --cluster-name=$(CLUSTER_NAME) \
+	     --gcp-project-id=$(GCP_PROJECT) \
+	     --gcp-region=us-east1 \
+	     --base-domain=roachlove.com \
+	     --pull-secret-file ~/Downloads/pull-secret.txt \
```
Comment (chrislovecnm) [inline Makefile:143]: TODO use variable for region, domain and pull secret file.

**Review by keith-mcclellan — APPROVED** [review]
LGTM

**Review by keith-mcclellan — DISMISSED** [review]
LGTM

**Review by keith-mcclellan — APPROVED** [review]
LGTM

**Pull request merged**